nautobot/extras/plugins/__init__.py

```diff
@@ -491,6 +518,106 @@ def register_filter_extensions(filter_extensions, plugin_name):
                 )
 
 
+#
+# Table Extensions
+#
+
+
+class TableExtension:
+    """Template class for extending Tables.
+
+    An app can override the default columns for a table by either:
+    - Extending the original default columns to include custom columns.
+        - add_to_default_columns = ("my_new_column",)
+    - Removing native columns from the default columns.
+        - remove_from_default_columns = ("tenant",)
+    """
+
+    model = None
+    table_columns = {}
+    add_to_default_columns = ()
+    remove_from_default_columns = ()
+
+    @classmethod
+    def alter_queryset(cls, queryset):
+        """Alter the View class QuerySet.
+
+        This is a good place to add `prefetch_related` to the view queryset.
+        example:
+            return queryset.prefetch_related("my_model_set")
+        """
+        return queryset
+
+
+def register_table_extensions(table_extensions, app_name):
+    """Register a list of TableExtension classes."""
+    for table_extension in table_extensions:
+        _validate_is_subclass_of_table_extension(table_extension)
+        _register_table_base_columns(table_extension, app_name)
+        _modify_default_table_columns(table_extension)
+        _alter_table_view_queryset(table_extension, app_name)
+
+
+def _register_table_base_columns(table_extension, app_name):
+    """Register each column in the TableExtension."""
+    from nautobot.core.utils.lookup import get_table_for_model
+
+    table = get_table_for_model(table_extension.model)
+    for name, column in table_extension.table_columns.items():
+        _validate_table_column_name_is_prefixed_with_app_name(name, app_name)
+        try:
+            _add_column_to_table_base_columns(table, name, column)
+        except AttributeError as error:
+            logger.error(*error.args)
+
+
+def _add_column_to_table_base_columns(table, column_name, column):
+    """Attach a column to an existing table."""
+    import django_tables2
```

Comment:
Does this need to be a local import here, or could it be moved to the top of the file?

Reply:
This needs to be a local import.  Otherwise we get this when trying to start the server or workers.
```
celery_worker-1  | Traceback (most recent call last):
celery_worker-1  |   File "/usr/local/bin/nautobot-server", line 3, in <module>
celery_worker-1  |     from nautobot.core.cli import main
celery_worker-1  |   File "/source/nautobot/core/cli/__init__.py", line 17, in <module>
celery_worker-1  |     from nautobot.extras.plugins.utils import load_plugins
celery_worker-1  |   File "/source/nautobot/extras/plugins/__init__.py", line 7, in <module>
celery_worker-1  |     import django_tables2
celery_worker-1  |   File "/usr/local/lib/python3.12/site-packages/django_tables2/__init__.py", line 19, in <module>
celery_worker-1  |     from .tables import Table, table_factory
celery_worker-1  |   File "/usr/local/lib/python3.12/site-packages/django_tables2/tables.py", line 186, in <module>
celery_worker-1  |     class Table(metaclass=DeclarativeColumnsMetaclass):
celery_worker-1  |   File "/usr/local/lib/python3.12/site-packages/django_tables2/tables.py", line 26, in __new__
celery_worker-1  |     attrs["_meta"] = opts = TableOptions(attrs.get("Meta", None), name)
celery_worker-1  |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker-1  |   File "/usr/local/lib/python3.12/site-packages/django_tables2/tables.py", line 121, in __init__
celery_worker-1  |     DJANGO_TABLES2_TEMPLATE = getattr(
celery_worker-1  |                               ^^^^^^^^
celery_worker-1  |   File "/usr/local/lib/python3.12/site-packages/django/conf/__init__.py", line 102, in __getattr__
celery_worker-1  |     self._setup(name)
celery_worker-1  |   File "/usr/local/lib/python3.12/site-packages/django/conf/__init__.py", line 89, in _setup
celery_worker-1  |     self._wrapped = Settings(settings_module)
celery_worker-1  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker-1  |   File "/usr/local/lib/python3.12/site-packages/django/conf/__init__.py", line 217, in __init__
celery_worker-1  |     mod = importlib.import_module(self.SETTINGS_MODULE)
celery_worker-1  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker-1  |   File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
celery_worker-1  |     return _bootstrap._gcd_import(name[level:], package, level)
celery_worker-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker-1  | ModuleNotFoundError: No module named 'nautobot_config'
```

Reply:
Ack. Didn't know that importing tables2 forced evaluation of the Django settings.